learningresources/tests/base.py

```diff
@@ -11,14 +11,17 @@
 
 class LoreTestCase(TestCase):
     """Handle often-needed things in tests."""
+    TEST_USERNAME = 'test'
+    TEST_PASSWORD = 'test'
 
     def setUp(self):
         """set up"""
         super(LoreTestCase, self).setUp()
-        self.username = 'test'
-        self.password = 'test'
-        self.user, _ = User.objects.get_or_create(username=self.username)
-        self.user.set_password(self.password)
-        self.user.save()
+        self.user = User.objects.create_user(
+            username=self.TEST_USERNAME, password=self.TEST_PASSWORD
```

Comment:
This works, but it's not technically correct. I read a great blog post that explained it really well a while back. I can't find it now, but [this one](https://timothyawiseman.wordpress.com/2012/10/06/class-and-instance-variables-in-python-2-7/) says pretty much the same thing.

The short version is that setting class variables is not the same as setting defaults for instance variables, and changing a value may not take effect in other places as intended, or may change things for other instances.

Short example to illustrate:

```python
class Thing(object):
    fred = "bicycle"

    def something(self):
        print "self:", self.fred
        print "thing:", Thing.fred
        print ""

b = Thing()

print "before"
b.something()

b.fred = "monkey"

print "after"
b.something()
```